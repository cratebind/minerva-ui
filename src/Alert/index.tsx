import React from 'react';
import styled from 'styled-components';
import { MinervaProps, systemProps, Flex } from '../layout';
import Icon from '../Icon';
// import Button from '../Button';
import { Text } from '../Text';
import { forwardRefWithAs } from '../type-utilities';

export const StyledAlert = styled(Flex)(
  {
    padding: '16px',
    borderRadius: '5px',
    alignItems: 'center',
    // justifyContent: 'space-between',
  },
  systemProps
);

export const StyledAlertTitle = styled(Text)(
  {
    fontWeight: 'bold',
    marginRight: '5px',
  },
  systemProps
);

export const StyledAlertDescription = styled(Text)``;

export const StyledCloseText = styled(Text)`
  text-decoration: underline;
`;

export const StyledAlertInner = styled(Flex)``;

export const StyledAlertIconSection = styled(Flex)`
  justify-content: flex-start;
`;

export interface AlertProps extends MinervaProps {
  children?: React.ReactNode;
  status?: 'error' | 'success' | 'warning' | 'info';
  title?: string;
  bg?: string;
  icon?: string;
  // isOpen?: boolean;
  // onClose?: any;
  // canBeClosed?: boolean;
  // hasCloseIcon?: boolean;
  // closeText?: string;
  props?: any;
}

// @TODO: Move these to the theme so they can be customized
const alertTypes = {
  error: {
    statusColor: '#f8b4b4',
    statusIcon: null,
  },
  success: {
    statusColor: '#bcf0da',
    statusIcon: null,
  },
  warning: {
    statusColor: '#fdf6b2',
    statusIcon: null,
  },
  info: {
    statusColor: '#c3ddfd',
    statusIcon: null,
  },
};

export const Alert = forwardRefWithAs<AlertProps, 'div'>(function Alert(
  { title, children, status, icon, ...props }: AlertProps,
  ref
) {
  const { statusColor, statusIcon } =
    // if status is provided and is a provided alertType, return it
    status && Object.keys(alertTypes).includes(status)
      ? alertTypes[status]
      : {
          statusColor: '#e5e7eb',
          statusIcon: null,
        };

  const alertIcon = icon || statusIcon;

  return (
    <StyledAlert
      ref={ref}
      role="alert"
      bg={statusColor}
      lineHeight={1}
      aria-label={`${title}`}
      aria-live="polite"
      fontFamily="body"
      {...props}
    >
      {alertIcon && <Icon name={alertIcon} size="20px" mr={2} />}
      <StyledAlertTitle>{title}</StyledAlertTitle>
      {children}
    </StyledAlert>
  );
});

export default Alert;
