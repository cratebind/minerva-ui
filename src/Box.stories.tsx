import React from 'react';
import { Meta, Story } from '@storybook/react';
import { Box, BoxProps, Stack } from '.';

const meta: Meta = {
  title: 'Box',
  component: Box,
  argTypes: {
    children: {
      control: {
        type: 'text',
      },
    },
  },
  parameters: {
    controls: { expanded: true },
  },
};

export default meta;

const Template: Story<BoxProps> = (args: Omit<BoxProps, 'as'>) => (
  <Box {...args} />
);

export const Default = Template.bind({});
Default.args = {
  children: 'Basic Box',
};

export const PseudoProps = () => (
  <Stack horizontal>
    <Box _hover={{ bg: 'red.500' }}>Hover Box</Box>
    <Box as="button" _focus={{ bg: 'red.500' }}>
      Focus Button
    </Box>
  </Stack>
);

// export const Variants = () => (
//   <Stack horizontal>
//     <Button variant="primary">Primary Button</Button>
//     <Button variant="secondary">Secondary Button</Button>
//     <Button variant="tertiary">Tertiary Button</Button>
//   </Stack>
// );

// export const LoadingButton = Template.bind({});
// LoadingButton.args = {
//   children: 'Basic Button',
//   isLoading: true,
// };

// export const DisabledButton = Template.bind({});
// DisabledButton.args = {
//   children: 'Basic Button',
//   disabled: true,
// };
