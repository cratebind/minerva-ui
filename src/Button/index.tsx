import React, { forwardRef } from 'react';
import warning from 'tiny-warning';
import Spinner from '../Spinner';
import PseudoBox, { PseudoBoxProps } from '../PseudoBox';
import { useComponentStyles, useTheme } from '../theme';
import { MinervaProps } from '../layout';

export const buttonVariants = {
  primary: {
    bg: 'indigo.800',
    borderColor: 'indigo.800',
    color: 'white',
    _hover: {
      bg: 'indigo.900',
      borderColor: 'indigo.900',
    },
  },
  secondary: {
    bg: 'white',
    borderColor: 'indigo.800',
    color: 'indigo.800',
    _hover: {
      bg: 'indigo.800',
      color: 'white',
    },
  },
  tertiary: {
    bg: 'white',
    borderColor: 'transparent',
    color: 'indigo.800',
    _hover: {
      textDecoration: 'underline',
    },
  },
};

export interface ButtonProps extends MinervaProps, PseudoBoxProps {
  children?: React.ReactNode;
  /** Toggles disabled pseudo class */
  disabled?: boolean;
  /** If `true`, button will show a spinner. */
  isLoading?: boolean;
  /** Accessibility label for buttons without text content */
  name?: string;
  /** Button variant styles inherited from theme */
  variant?: string;
  /** HTML Button Type (https://developer.mozilla.org/en-US/docs/Web/HTML/Element/button) */
  type?: 'button' | 'reset' | 'submit';
}

export const Button = forwardRef(function Button(
  {
    children,
    disabled = false,
    as: Comp = 'button',
    isLoading = false,
    name,
    variant,
    ...props
  }: ButtonProps,
  ref
) {
  const theme = useTheme();

  // if a variant is provided and it doesn't exist in the current theme, warn in development
  warning(
    !variant ||
      (variant &&
        theme?.variants?.Button &&
        Object.keys(theme?.variants?.Button).includes(variant)),
    `Variant "${variant}" not found in theme variants for <Button />:\n\n${theme
      ?.variants?.Button &&
      `Expected one of:\n[${Object.keys(theme.variants.Button).join(', ')}]`}`
  );

  warning(
    children || (!children && name),
    'Buttons without children require a `name` attribute to be accessible.'
  );

  const variantStyles =
    variant && theme?.variants?.Button ? theme.variants.Button[variant] : {};

  const componentStyles = useComponentStyles('Button');

  return (
    <PseudoBox
      ref={ref}
      as={Comp}
      disabled={disabled || isLoading}
      role="button"
      cursor="pointer"
      _disabled={{
        cursor: 'not-allowed',
      }}
      aria-busy={isLoading}
      name={name}
      {...componentStyles}
      {...variantStyles}
      {...props}
    >
      {isLoading ? <Spinner /> : children}
    </PseudoBox>
  );
});

export default Button;

// if (__DEV__) {
//   Button.propTypes = exact({
//     ...minervaPropTypes,
//     children: PropTypes.node,
//     disabled: PropTypes.bool,
//     isLoading: PropTypes.bool,
//     onClick: PropTypes.func,
//   });
// }
