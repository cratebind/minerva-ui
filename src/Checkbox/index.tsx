import React, { forwardRef } from 'react';
import { CustomCheckboxContainer, CustomCheckboxInput } from '@reach/checkbox';
import { MinervaProps, Box } from '../layout';
import PseudoBox from '../PseudoBox';
import { useComponentStyles } from '../theme';

type BaseProps = MinervaProps & React.InputHTMLAttributes<HTMLInputElement>;

export interface CustomCheckboxProps {
  children?: React.ReactNode;
  checkColor?: string;
}

export type CheckboxProps = CustomCheckboxProps & BaseProps;

export const Checkbox = forwardRef(function Checkbox(
  {
    children,
    onChange,
    disabled,
    checked,
    defaultChecked,
    checkColor = '#fff',
    ...props
  }: CheckboxProps,
  ref: any
) {
  const componentStyles = useComponentStyles('Checkbox');
  return (
    <label>
      <Box
        as={CustomCheckboxContainer}
        display="flex"
        alignItems="center"
        onChange={onChange}
        checked={checked}
        disabled={disabled}
        defaultChecked={defaultChecked}
      >
        <PseudoBox
          as={CustomCheckboxInput}
          backgroundImage={`url("data:image/svg+xml,%3Csvg xmlns='http://www.w3.org/2000/svg' width='192' height='192' fill='%23000000' viewBox='0 0 256 256'%3E%3Crect width='256' height='256' fill='none'%3E%3C/rect%3E%3Cpolyline points='216 72.005 104 184 48 128.005' fill='none' stroke='${checkColor.replace(
            '#',
            '%23'
          )}' stroke-linecap='round' stroke-linejoin='round' stroke-width='16'%3E%3C/polyline%3E%3C/svg%3E");`}
          backgroundSize="100%"
          backgroundPosition="center"
          backgroundRepeat="no-repeat"
          type="checkbox"
          ref={ref}
          appearance="none"
          width="18px"
          height="18px"
          marginRight="8px"
          borderRadius="4px"
          borderWidth="1px"
          borderStyle="solid"
          padding="2px"
          backgroundColor="#fff"
          transition="background-color, border-color 150ms cubic-bezier(0.4, 0, 0.2, 1)"
          _checked={{
            backgroundColor: '#5850ec',
            borderColor: '#5850ec',
          }}
          _focus={{
            borderColor: '#a4cafe',
            boxShadow: '0 0 0 3px rgba(118,169,250,.45)',
            outline: 0,
          }}
          {...componentStyles}
          {...props}
        />
        {children}
      </Box>
    </label>
  );
});

export default Checkbox;
