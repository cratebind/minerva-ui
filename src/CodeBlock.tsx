import React, { useState } from 'react';
// import lightTheme from 'prism-react-renderer/themes/nightOwlLight';
import darkTheme from 'prism-react-renderer/themes/nightOwl';
import { LiveProvider, LiveEditor, LiveError, LivePreview } from 'react-live';
import { ProviderWithIcons } from '../.storybook/preview';
import { mdx } from '@mdx-js/react';
import * as Minerva from '.';
import useClipboard from './hooks/useClipboard';

const { Block } = Minerva;

export const liveEditorStyle = {
  fontSize: 14,
  // overflowX: "auto",
  fontFamily: 'Menlo,monospace',
  borderRadius: 10,
  // padding: 20,
};

const wrapperStyles = {
  backgroundColor: 'rgb(1, 22, 39)',
  borderRadius: '10px',
  padding: '20px',
  marginBottom: '32px',
  marginTop: '32px',
};

// @TODO: Figure out why "position: relative" causes TS error
const Wrapper = ({ children }) => (
  <Block {...wrapperStyles} style={{ position: 'relative' }}>
    {children}
  </Block>
);

// const highlightStyle = {
//   padding: 20,
//   fontSize: 14,
//   overflow: "auto",
//   lineHeight: "1.5",
//   fontFamily: "Menlo,monospace",
// };

export const liveErrorStyle = {
  fontFamily: 'Menlo, monospace',
  fontSize: 14,
  padding: '1em',
  // overflowX: "auto",
  color: 'rgb(220, 53, 69)',
  backgroundColor: 'rgb(1,22,39)',
  borderRadius: 10,
  marginTop: '10px',
  // backgroundColor: 'red',
};

// const Box = ({ children, style }: any) => <div style={style}>{children}</div>

const previewStyles = {
  fontFamily:
    '-apple-system,".SFNSText-Regular","San Francisco",BlinkMacSystemFont,"Segoe UI","Helvetica Neue",Helvetica,Arial,sans-serif',
  border: '1px solid rgb(226, 232, 240)',
  borderRadius: 4,
  padding: 20,
};

const LiveCodePreview = (props: any) => (
  <div style={previewStyles}>
    <ProviderWithIcons>
      <Minerva.GlobalStyles />
      <LivePreview {...props} />
    </ProviderWithIcons>
  </div>
);

const CopyButton = (props: any) => (
  <Minerva.Button
    position="absolute"
    fontWeight="600"
    fontSize="12px"
    top="15px"
    padding="0 12px"
    textTransform="uppercase"
    fontFamily={`-apple-system, BlinkMacSystemFont, "Segoe UI", Helvetica, Arial, sans-serif, "Apple Color Emoji", "Segoe UI Emoji", "Segoe UI Symbol";`}
    zIndex="1"
    right="1.25em"
    style={{
      textTransform: 'uppercase',
    }}
    {...props}
  />
);

const Provider = ({ children, ...props }) => (
  <Minerva.ThemeProvider>
    {/* <Minerva.GlobalStyles /> */}
    <LiveProvider {...props}>{children}</LiveProvider>
  </Minerva.ThemeProvider>
);

const CodeBlock = ({
  className,
  live = true,
  isManual,
  render,
  children,
  ...props
}: any) => {
  const [editorCode, setEditorCode] = useState(children.trim());

  const language = className && className.replace(/language-/, '');
  const { onCopy, hasCopied } = useClipboard(editorCode);

  // const { colorMode } = useColorMode();
  const themes = {
    // light: lightTheme,
    dark: darkTheme,
  };
  const theme = themes['dark'];

  const liveProviderProps = {
    theme,
    language,
    code: editorCode,
    // transformCode: code => "/** @jsx mdx */" + code,
    scope: {
      // ...ReactIcons,
      ...Minerva,
      mdx,
      // StarIcon,
      // FocusLock,
      // ChakraPortal,
    },
    noInline: isManual,
    ...props,
  };

  const handleCodeChange = (newCode: any) => setEditorCode(newCode.trim());

  if (language === 'jsx' && live === true) {
    return (
      <Provider {...liveProviderProps}>
        <LiveCodePreview />
        <Wrapper>
          <LiveEditor onChange={handleCodeChange} style={liveEditorStyle} />
          <CopyButton onClick={onCopy}>
            {hasCopied ? 'copied' : 'copy'}
          </CopyButton>
          {/* <EditableNotice /> */}
        </Wrapper>
        <LiveError style={liveErrorStyle} />
      </Provider>
    );
  }

  if (render) {
    return (
      <div style={{ marginTop: '40px' }}>
        <Provider {...liveProviderProps}>
          <LiveCodePreview />
        </Provider>
      </div>
    );
  }

  return (
    <Provider disabled {...liveProviderProps}>
      <Wrapper>
        <LiveEditor style={liveEditorStyle} />

        <CopyButton top="1.25em" onClick={onCopy}>
          {hasCopied ? 'copied' : 'copy'}
        </CopyButton>
      </Wrapper>
    </Provider>
  );
};

CodeBlock.defaultProps = {
  mountStylesheet: false,
};

export default CodeBlock;
