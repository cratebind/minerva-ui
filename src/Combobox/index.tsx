import React, { InputHTMLAttributes, ReactNode } from 'react';
import {
  Combobox as ReachCombobox,
  ComboboxInput as ReachComboboxInput,
  ComboboxPopover as ReachComboboxPopover,
  ComboboxList as ReachComboboxList,
  ComboboxOption as ReachComboboxOption,
  ComboboxOptionText as ReachComboboxOptionText,
  ComboboxProps,
  ComboboxInputProps,
  ComboboxPopoverProps,
  ComboboxListProps,
  ComboboxOptionProps,
} from '@reach/combobox';
// import css from '@styled-system/css';

import Input from '../Input';
import { OverlayBox } from '../Menu';
import { Box, MinervaProps } from '../layout';
// import { Box } from 'react-feather';

export const Combobox = (props: ComboboxProps) => <ReachCombobox {...props} />;

export const ComboboxInput = (
  props: ComboboxInputProps & InputHTMLAttributes<HTMLInputElement>
) => <Input as={ReachComboboxInput} {...props} />;

export const ComboboxPopover = (props: ComboboxPopoverProps & MinervaProps) => (
  <OverlayBox
    mt="8px"
    borderRadius="6px"
    border={0}
    overflow="hidden"
    zIndex="50"
    as={ReachComboboxPopover}
    data-testid="combobox-popover"
    {...props}
  />
);

export const ComboboxList = (
  props: ComboboxListProps & { children?: ReactNode }
) => <ReachComboboxList {...props} />;

export const ComboboxOption = (props: ComboboxOptionProps & MinervaProps) => (
  <Box
    as={ReachComboboxOption}
    cursor="pointer"
    padding="0.5rem 1rem"
    css={{
      '&:hover,&[aria-selected="true"]': {
        backgroundColor: '#f4f5f7',
        color: '#161e2e',
      },
    }}
    {...props}
  />
);

export const ComboboxOptionText = (props: MinervaProps) => (
  <ReachComboboxOptionText {...props} />
);
