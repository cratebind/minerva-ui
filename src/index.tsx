// import * as Icon from 'react-feather';

import { ChevronDown, Close } from './Icon/baseIcons';

// // https://gist.github.com/thevangelist/8ff91bac947018c9f3bfaad6487fa149
// function toKebabCase(string) {
//   return string
//     .replace(/([a-z])([A-Z])/g, '$1-$2')
//     .replace(/\s+/g, '-')
//     .toLowerCase();
// }

// export const defaultIcons: { [key: string]: any } = Object.keys(Icon).reduce(
//   (result, iconName) => {
//     result[toKebabCase(iconName)] = Icon[iconName];
//     return result;
//   },
//   {}
// );

// export const defaultIcons = baseIcons;
export const defaultIcons = {
  'chevron-down': ChevronDown,
  close: Close,
};

/**
 * Utilities
 */

export { default as ThemeProvider } from './ThemeProvider';
export { default as VisuallyHidden } from './VisuallyHidden';
export {
  default as GlobalStyles,
  AnimationStyles,
  ReachGlobalStyles,
  CSSReset,
} from './GlobalStyles';
// export { useTheme, useComponentStyles } from './theme';
export { default as baseTheme } from './theme';
export { default as defaultTheme } from './theme';
export * from './theme';
export { default as styled } from 'styled-components';

/**
 * Display
 */
export * from './Text';
export * from './Link';
export * from './Heading';
export * from './Tag';
export * from './Alert';
export * from './Tooltip';
export * from './Accordion';

/**
 * Layout
 */
export * from './layout';
export { default as PseudoBox } from './PseudoBox';
export * from './PseudoBox';
export * from './Stack';

/**
 * Inputs
 */
export * from './Button';
export * from './Checkbox';
export * from './Switch';
export * from './Input';
export * from './Input';
export * from './Combobox';
export * from './Menu';
export * from './Radio';
export * from './InputField';
export { default as InputField } from './InputField';
export * from './Select';
export * from './TagsInput';
export * from './Tabs';

/**
 * Elements
 */
export * from './Image';
export * from './Table';
export * from './Icon';
export * from './Table';
export * from './Modal';
export * from './Drawer';
export * from './Skeleton';
export * from './Skeleton';

/**
 * Helpers
 */
export * from './Spinner';

/**
 * Utilities
 */

export * from './useClipboard';
export { default as useClipboard } from './useClipboard';
export * from './useDisclosure';
export { default as useDisclosure } from './useDisclosure';
export * from './useMedia';
export { default as useMedia } from './useMedia';
export * from './useNetwork';
export { default as useNetwork } from './useNetwork';
export * from './useLocalStorage';
export { default as useLocalStorage } from './useLocalStorage';
