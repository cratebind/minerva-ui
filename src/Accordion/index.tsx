import React, { forwardRef } from 'react';
// import styled from 'styled-components';
import {
  Accordion as ReachAccordion,
  AccordionProps as ReachAccordionProps,
  AccordionItem as ReachAccordionItem,
  AccordionItemProps as ReachAccordionItemProps,
  AccordionButton as ReachAccordionButton,
  AccordionPanel as ReachAccordionPanel,
  AccordionPanelProps as ReachAccordionPanelProps,
} from '@reach/accordion';
import { useComponentStyles } from '../theme';
import { MinervaProps, Box } from '../layout';
import Button from '../Button';

export interface AccordionProps extends MinervaProps {
  children?: React.ReactNode;
}

export interface AccordionButtonProps extends MinervaProps {
  children?: React.ReactNode;
}

export interface AccordionPanelProps extends MinervaProps {
  children?: React.ReactNode;
}

export type AllAccordionItemProps = ReachAccordionItemProps & MinervaProps;

export const AccordionItem = forwardRef(function AccordionItem(
  { ...props }: AllAccordionItemProps,
  ref
) {
  return <Box as={ReachAccordionItem} ref={ref} {...props} />;
});

export const AccordionButton = forwardRef(function AccordionButton(
  { children, ...props }: AccordionButtonProps,
  ref
) {
  const componentStyles = useComponentStyles('AccordionButton');

  return (
    <ReachAccordionButton
      as={Button}
      borderColor="transparent"
      width="100%"
      flexDirection="row"
      justifyContent="space-between"
      alignItems="flex-start"
      bg="white"
      py="12px"
      fontSize="18px"
      fontWeight="400"
      borderTopColor="gray.300"
      borderRadius={0}
      ref={ref}  
      {...componentStyles}
      {...props}
    >
      {children}
      {/* TODO Need to figure out how to determine if Item is expanded and rotate icon */}
      {/* <Icon name="chevron-down" ml={2} size="14px" /> */}
    </ReachAccordionButton>
  );
});

export type ExtendedAccordionPanelProps = ReachAccordionPanelProps &
  AccordionPanelProps;

export const AccordionPanel = forwardRef(function AccordionPanel(
  { ...props }: ExtendedAccordionPanelProps,
  ref
) {
  return (
    <Box
      as={ReachAccordionPanel}
      px="14px"
      pr="23%"
      py="14px"
      width="100%"
      borderTopWidth="1px"
      borderTopColor="gray.300"
      ref={ref}
      {...props}
    />
  );
});

export type AllAccordionProps = ReachAccordionProps & MinervaProps;
export const Accordion = forwardRef(function Alert(
  { children, ...props }: AllAccordionProps,
  ref
) {
  return (
    <Box
      as={ReachAccordion}
      borderBottomColor="gray.300"
      borderBottomWidth="1px"
      {...props}
      ref={ref}
    >
      {children}
    </Box>
  );
});

export default Accordion;
