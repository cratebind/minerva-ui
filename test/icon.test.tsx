import React from 'react';
import { render } from '@testing-library/react';
import 'jest-styled-components';
import { defaultIcons, ThemeProvider } from '../src';

const Close = defaultIcons['close'];
const ChevronDown = defaultIcons['chevron-down'];

describe('Default Icons', () => {
  it('should render', () => {
    const { container } = render(
      <ThemeProvider>
        <Close />
        <ChevronDown />
      </ThemeProvider>
    );
    expect(container).toMatchSnapshot();
  });
});
