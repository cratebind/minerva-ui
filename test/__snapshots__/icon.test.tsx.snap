// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Default Icons should render 1`] = `
<div>
  <svg
    color="currentColor"
    fill="currentColor"
    height="24"
    viewBox="0 0 256 256"
    width="24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <rect
      fill="none"
      height="256"
      width="256"
    />
    <line
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="16"
      x1="200"
      x2="56"
      y1="56"
      y2="200"
    />
    <line
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="16"
      x1="200"
      x2="56"
      y1="200"
      y2="56"
    />
  </svg>
  <svg
    fill="none"
    height="7"
    viewBox="0 0 10 7"
    width="10"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M4.95174 6.54518L0.70874 2.30218L2.12374 0.888184L4.95174 3.71718L7.77974 0.888184L9.19474 2.30218L4.95174 6.54518Z"
      fill="black"
    />
  </svg>
</div>
`;
